Simplify file upload interface

staab opened the pull request:
@cesardeazevedo there are a bunch of loose ends in this PR, but you should be able to see where I'm going with it. Basically, we don't need uploadType/uploadUrl, built in uploaders, or hash and sign functions if we just give the user a `upload` function that returns an UploadTask. Obviously, this is not a backwards compatible change, but if you're ok with that I can finish up the PR.

cesardeazevedo commented on src/uploaders/nip96.ts:
```diff
@@ -1,32 +0,0 @@
-  sign?: (event: EventTemplate) => Promise<NostrEvent> | NostrEvent
-}
-
-export async function uploadNIP96(options: NIP96Options): Promise<UploadTask> {
```
the library still should provide these functions no?

staab replied:
Well, similar to the encryption functions they're now only being called by the application code. There are plenty of implementations of blossom/nip96 out there, and the impetus for this PR was that the blossom uploader no longer worked for my use case. I can leave them in if you want, but it seems better to put them in nostr-tools or somewhere else instead.

staab commented:
Ok, I've got this in pretty good shape. Summary of changes:

- I've removed uploadUrl/uploadType from file/video extensions. Different upload methods for different files can easily be implemented by the user of the library via the `upload` option.
- I've removed UploadChip for the same reason, a custom node view can be used by the user of the library to implement similar behavior outside of the extension if desired. I think this will be a lot nicer DX too, since tiptap code is pretty nasty to work with.
- I added `ox` and `size` to the default tags that get generated.
- I renamed `uploadError` to `error` since the context should be obvious.
- I removed `msw` since there's no need to test upload functions any more.
- I removed the blossom/nip96 uploaders. We could restore these if you want, but I think it's nice to keep the library small and focused since they're not hard to implement, and (as my recent use case demonstrated) might need to be tweaked per implementation.

staab commented:
Here's my library implementation of blossom: https://github.com/coracle-social/welshman/blob/master/packages/util/src/Blossom.ts
And here's my usage of it: https://github.com/coracle-social/flotilla/blob/dev/src/app/editor/index.ts#L73

Pretty nasty honestly, but mostly because of the file extension appending and having to sign the auth event. Lots of mistakes to make here, so a default implementation of at least blossom would probably be good to have. What do you think?

cesardeazevedo commented:
It's overall looking really good, just need a bit of more time to test it

cesardeazevedo commented:
Thanks so much for this, for even updating the docs a bit, released at `1.0.0`
